### DISCUSSION TITLE
I want to reduce the transfer volume of Storage egress.

### DISCUSSION THREAD

**Discussion by ru88s** [supabase/supabase#8917] [Questions] [answered]
Is it possible to reduce the amount of transfer by adjusting the cache time?
I would like to know if there is a good way to do this.

![スクリーンショット 2022-09-12 15 54 07](https://user-images.githubusercontent.com/42270511/189591169-6867eeed-1e34-4bc2-b73d-119b1d0493df.png)

**Comment by hjaber** [comment]
Could you provide more context of your storage? What type of objects are they (images, pdfs, etc) and how are they consumed on your site?

**Reply by ru88s** [reply]
Image only.
I run a bulletin board site.

**Comment by hjaber** [comment, marked as answer]
Without knowing more it is difficult to say if this is expected or not but your egress seems high. I would assume with a cloudfare CDN that your egress would be lower. 

Could you post a response header of a image on your site? 

![image](https://user-images.githubusercontent.com/41503068/189808497-d9fc84f8-5ccf-47a7-b856-bc4e07d390ce.png)

**Reply by ru88s** [reply]
Image via Cloudflare.

![スクリーンショット 2022-09-13 20 14 48](https://user-images.githubusercontent.com/42270511/189887347-56e413dc-4ad0-401d-8387-e66e796d7824.png)

**Reply by hjaber** [reply]
> Image via Cloudflare.
> 
> ![スクリーンショット 2022-09-13 20 14 48](https://user-images.githubusercontent.com/42270511/189887347-56e413dc-4ad0-401d-8387-e66e796d7824.png)

It is correctly being served from the Cloudfare CDN. The `cache-control: max-age=3600` is lasting 60 minutes,  a user's browser will request the image again if they view the same image after 60 minutes.

I'd recommend extending your cache-control from 60 minutes to 365 days upon upload and seeing if that improves:
```
const { data, error } = await supabase.storage
  .from('avatars')
  .upload('public/avatar1.png', avatarFile, {
    cacheControl: '31536000',
  })
```

I'm not sure how to apply this your already uploaded images but can significantly improve your future image uploads.

**Reply by ru88s** [reply]
Thank you very much.
Cache time has been adjusted.
I will see how this works.

![スクリーンショット 2022-09-14 13 48 58](https://user-images.githubusercontent.com/42270511/190062103-7780fddd-fb1a-493c-add8-133dea9d78a8.png)

**Reply by hjaber** [reply]
Fingers crossed it saves you some transfer egress! It helped me a lot initially. 

Please update this thread in a month if you can reporting if it helped so others can gain insight as well.

**Reply by ru88s** [reply]
I will report back to you. Thank you!

**Reply by hjaber** [reply]
> I will report back to you. Thank you!

Any update?

**Reply by francistogram** [reply]
@ru88s did this end up working? I'm trying to solve something similar
